function-aws-api-proxy/src/main/java/io/micronaut/function/aws/proxy/MicronautLambdaContainerHandler.java

```diff
@@ -399,48 +414,152 @@ private MicronautAwsProxyResponse<?> toAwsProxyResponse(
+        if (bodyArgument == null) {
+            JsonNode encodedValues = lambdaContainerEnvironment.getObjectMapper().valueToTree(formUrlEncodedBodyToConvertibleValues(body));
+            return Optional.ofNullable(encodedValues);
+        }
+        if (nestedBody(bodyArgument)) {
+            return Optional.ofNullable(formUrlEncodedBodyToConvertibleValues(body));
+        }
+        return bindFormUrlEncoded(bodyArgument, body);
+    }
+
+    @NonNull
+    private Optional<Object> getJsonDecodedBody(@Nullable Argument<?> bodyArgument,
+                                                @NonNull String body) {
+
+        if (bodyArgument == null) {
+            JsonMediaTypeCodec jsonCodec = lambdaContainerEnvironment.getJsonCodec();
+            JsonNode decoded = jsonCodec.decode(JsonNode.class, body);
+            return Optional.of(decoded);
+        }
+        JsonMediaTypeCodec jsonCodec = lambdaContainerEnvironment.getJsonCodec();
+        if (nestedBody(bodyArgument)) {
+            return Optional.of(new ConvertibleValuesMap(jsonCodec.decode(Argument.of(Map.class), body)));
+        }
+        return Optional.of(jsonCodec.decode(bodyArgument, body));
+    }
+
+    @NonNull
+    private static Optional<Map<String, List<String>>> formUrlEncodedBodyToMap(@NonNull String body) {
+        QueryStringDecoder decoder = new QueryStringDecoder(body, false);
+        Map<String, List<String>> parameters = decoder.parameters();
+        return CollectionUtils.isEmpty(parameters) ? Optional.empty() :
+            Optional.of(parameters);
+    }
+
+    @Nullable
+    private static ConvertibleValues<?> formUrlEncodedBodyToConvertibleValues(@NonNull String body) {
+        return formUrlEncodedBodyToMap(body)
+            .map(ConvertibleValuesMap::new)
+            .orElse(null);
+    }
+
+    private static boolean nestedBody(@NonNull Argument<?> bodyArgument) {
+        AnnotationMetadata annotationMetadata = bodyArgument.getAnnotationMetadata();
+        if (annotationMetadata.hasAnnotation(Body.class)) {
+            return annotationMetadata.stringValue(Body.class).isPresent();
+        }
+        return false;
+    }
+
+    @NonNull
+    private Optional<Object> bindFormUrlEncoded(@NonNull Argument<?> argument, @NonNull String formUrlEncodedString) {
+        return formUrlEncodedBodyToMap(formUrlEncodedString)
+            .flatMap(parameters -> bindFormUrlEncoded(argument, parameters));
+    }
+
+    @NonNull
+    private Optional<Object> bindFormUrlEncoded(@NonNull Argument<?> argument, @NonNull Map<String, List<String>> bodyParameters) {
+        Map<CharSequence, Object> source = new HashMap<>();
+        for (Map.Entry<String, List<String>> entry : bodyParameters.entrySet()) {
+            source.put(entry.getKey(), entry.getValue());
+        }
+        try {
+            return Optional.of(beanPropertyBinder.bind(argument.getType(), source));
+        } catch (ConversionErrorException e) {
+            return Optional.empty();
```

Comment:
Swallowing this might be undesirable, what if the error needs to be returned to the user or the developer is trying to understand why binding failed?

Reply:
I have added a log statement https://github.com/micronaut-projects/micronaut-aws/pull/1414/commits/515e34ac46145709bfb7e80def3770cdffea8d51